vllm/worker/model_runner.py

```diff
@@ -561,30 +597,49 @@ def _prepare_model_input(
                                            device=self.device)
 
         if self.attn_backend.get_name() == "flashinfer":
-            if not hasattr(self, "flashinfer_workspace_buffer"):
-                # Allocate 16MB workspace buffer
-                # Follow the example of flashinfer: https://docs.flashinfer.ai/api/python/decode.html
-                self.flashinfer_workspace_buffer = torch.empty(
-                    16 * 1024 * 1024, dtype=torch.uint8, device=self.device)
-            paged_kv_indptr_tensor = torch.tensor(paged_kv_indptr,
-                                                  dtype=torch.int,
-                                                  device=self.device)
-            paged_kv_indices_tensor = torch.tensor(paged_kv_indices,
-                                                   dtype=torch.int,
-                                                   device=self.device)
-            paged_kv_last_page_len_tensor = torch.tensor(
-                paged_kv_last_page_len, dtype=torch.int, device=self.device)
+            if len(paged_kv_indptr) > 0:
+                paged_kv_indices_tensor = torch.tensor(paged_kv_indices,
+                                                       device='cpu',
+                                                       dtype=torch.int)
+                paged_kv_indptr_tensor = torch.tensor(paged_kv_indptr,
+                                                      device='cpu',
+                                                      dtype=torch.int)
+                paged_kv_last_page_len_tensor = torch.tensor(
+                    paged_kv_last_page_len, device='cpu', dtype=torch.int)
+            else:
+                paged_kv_indices_tensor = None
+                paged_kv_indptr_tensor = None
+                paged_kv_last_page_len_tensor = None
+
+            if num_decode_tokens:
+                if use_captured_graph and not is_profile_run:
+                    self.decode_wrapper = self.graph_runners[
+                        batch_size].decode_wrapper
+                elif self.decode_wrapper is None:
+                    self.decode_workspace_buffer = torch.empty(
+                        128 * 1024 * 1024,
```

Comment:
can you comment reasoning for the size of the buffer (128mb)?

Reply:
I got it from flashinfer tests 😂
https://github.com/flashinfer-ai/flashinfer/blob/main/python/tests/test_batch_decode_kernels.py#L164